src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -2776,18 +2781,32 @@ public void onDeleteWiki() {
       );
   }
 
+  private String getIconHTML() {
+    EntityTypeIcon icon = ginInjector.getEntityTypeIcon();
+    icon.configure(entityBundle.getEntityType());
+    // Pull the HTML out of the icon React component and inline it
+    // It's possible ReactDOM has not injected HTML to the widget yet, so use flushSync to force it to be written
+    ReactDOM.flushSync();
+    return icon.getElement().getInnerHTML();
+  }
```

Comment:
Super hacky, but it saves us some reimplementation work--we can pass HTML `showConfirmDeleteDialog`. So create a React component with the icon, pull out the HTML, and stick the HTML into the message.